File: core/src/main/java/kr/scripton/business/core/auth/LocalNonceCheckService.java
package kr.scripton.business.core.auth;

import java.util.HashSet;

public class LocalNonceCheckService implements NonceCheckService {
	private HashSet<String> set = new HashSet<>();

	@Override
	public Boolean isDuplicate(String nonce) {
		if (set.contains(nonce)) {
			return false;
		}
		set.add(nonce);
		return true;
	}
}

File: core/src/main/kotlin/kr/scripton/business/core/auth/LocalNonceCheckService.kt
package kr.scripton.business.core.auth

class LocalNonceCheckService : NonceCheckService {
    private val set = HashSet<String>()
    override fun isDuplicate(nonce: String): Boolean {
        if(set.contains(nonce)){
            return false
        }
        set.add(nonce)
        return true
    }

}